src/services/item/item.repository.ts

```diff
@@ -281,35 +290,24 @@ export class ItemRepository {
       andConditions.push(inArray(items.type, types));
     }
 
-    // TODO: enable back
-    // const allKeywords = params?.keywords?.filter((s) => s && s.length);
-    // if (allKeywords?.length) {
-    //   const keywordsString = allKeywords.join(' ');
-
-    //   // search in english by default
-    //   const matchEnglishSearchCondition = sql`${items.searchDocument} @@ plainto_tsquery('english', ${keywordsString})`;
-
-    //   // no dictionary
-    //   const matchSimpleSearchCondition = sql`${items.searchDocument} @@ plainto_tsquery('simple', ${keywordsString})`;
-
-    //   // raw words search
-    //   const matchRawWordSearchConditions = allKeywords.map((k) => ilike(items.name, `%${k}%`));
-
-    //   const searchConditions = [
-    //     matchEnglishSearchCondition,
-    //     matchSimpleSearchCondition,
-    //     ...matchRawWordSearchConditions,
-    //   ];
-
-    //   // search by member lang
-    //   const memberLang = actor && isMember(actor) ? actor?.lang : DEFAULT_LANG;
-    //   if (memberLang && ALLOWED_SEARCH_LANGS[memberLang]) {
-    //     const matchMemberLangSearchCondition = sql`${items.searchDocument} @@ plainto_tsquery(${ALLOWED_SEARCH_LANGS[memberLang]}, ${keywordsString})`;
-    //     searchConditions.push(matchMemberLangSearchCondition);
-    //   }
-
-    //   andConditions.push(or(...searchConditions));
-    // }
+    // keyword search
+    const allKeywords = params?.keywords?.filter((s) => s && s.length);
+    if (allKeywords && allKeywords.length) {
+      const keywordsString = allKeywords.join(' ');
+
+      andConditions.push(
+        or(
+          // search without lang
+          itemFullTextSearch(items, 'simple', keywordsString),
+          // search in english
+          itemFullTextSearch(items, 'english', keywordsString),
```

Comment:
Why is english treated as a special language that is always included ? Is this why we return undefined if the user has the default lang set ? 

I am not sure I understand the reason why english should be treated differently.

Reply:
Actually we use english for default for kinda "item lang". In the case the user has german as language, but is browsing an english item, the search will only happen in german if we don't add "english". But I will refactor to take into account the item's lang instead of the english only.